bin/report

```diff
@@ -18,7 +18,13 @@ while IFS= read -r line; do
   elif [[ $line == *deb ]]; then
     custom_packages+=("${line}")
   else
-    packages+=("${line}")
+    # while this is not documented behavior, the Aptfile format technically
+    # does allow for multiple packages separated by spaces to be specified
+    # on a single line due to how the download command is implemented
+    IFS=' ' read -ra package_names <<< "${line}"
+    for package_name in "${package_names[@]}"; do
+      packages+=("${package_name}")
+    done
```

Comment:
```suggestion
    # so we perform regular splitting on whitespace into array elements
    packages+=( $line )
```

You _do_ want simple word splitting by `IFS` here - this will take care of spaces and tabs (not quotes, but that's not handled by the `IFS=' ' read …` approach either, for which you would need, but definitely not want to use, `eval`).

Reply:
@dzuelke I tried you're suggestion. It does pass my test case but gets highlighted by `shellcheck` as a potential error with [SC2206](https://github.com/koalaman/shellcheck/wiki/SC2206). At this stage, since I'm only concerned about splitting on whitespace (quotes are handled later during `output_key_value()`) are you saying that I should prefer the more explicit `IFS=" " read -ra ...` form?

Reply:
Image `line=$'foo\tbar'`, that gets split in the download command, but not by your reader. So you'd at least have to change the `read` line to `IFS=$' \t'` to handle tabs as well.

The SC rule does make a good point about globbing though. The download command is actually affected by this: if you have an empty file called `nano`, and a line in `Aptfile` that says `na*`, it'll execute `apt-get install na*`, which will expand `na*` to `nano`.

All of this makes me wonder if we shouldn't instead pull the install info from Apt's `history.log`?

Example, for two separate installs of `nano` and `libimath-dev`:

```
Start-Date: 2024-03-12  14:29:36
Commandline: apt-get install nano
Install: nano:amd64 (6.2-1)
End-Date: 2024-03-12  14:29:36

Start-Date: 2024-03-12  14:36:04
Commandline: apt-get install libimath-dev
Install: libimath-3-1-29:amd64 (3.1.4-3, automatic), libimath-dev:amd64 (3.1.4-3)
End-Date: 2024-03-12  14:36:04
```

I'm not quite sure where Apt writes this to with the special options we pass it, but worth a look?

Reply:
@dzuelke I'd rather parse a structured file format we define (`Aptfile`) vs. a potentially unstructured log file (`history.log`).  If we find users have been adding glob patterns to their `Aptfile` then we can revisit.

I've changed `IFS=' '` to `IFS=$' \t'` as you suggested and tested it with a multi-package line containing tabs and that seems to work well enough.

Reply:
Alright, fine, but then let's fix `bin/compile` to behave the same: https://github.com/heroku/heroku-buildpack-apt/pull/115

Reply:
> If we find users have been adding glob patterns to their `Aptfile` then we can revisit.

@colincasey @dzuelke There are users that use globs (this was news to me too), eg:
https://ui.honeycomb.io/heroku/datasets/builds/result/JyttX69CKeN

Does that affect the choices here? (I've not had time to read the code; just raising awareness in case)

Reply:
Not for the instrumentation, no. We're capturing the glob correctly and it's providing insight into how the Aptfile is currently being used.

Reply:
No, this does not affect this. The globs still work fine, they just no longer expand anything that's in the CWD.

Before, if you had `mysql-client-*` in your `Aptfile`, due to Bash's default behavior of leaving unmatched globs, this would get passed to `apt-get install` as a literal `mysql-client-*`. Apt then expands that against its list of known package names that match this pattern.

However, if you then by accident added a file named `mysql-client-test.conf`, the result would be `apt-get install mysql-client-test.conf`.

Since #115, the glob is not expanded by the shell anymore, so it's now always passed literally, and Apt expands it as intended.